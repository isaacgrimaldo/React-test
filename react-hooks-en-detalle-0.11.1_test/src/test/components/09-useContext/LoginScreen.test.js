import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';

import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Prueba del LoginScreen.js', () => {
     
    const testUser = {
        id: 123,
        name: 'Fernando'
    };
    const setUser = jest.fn(() => {testUser});
    
    const wrapper = mount(
        <UserContext.Provider
          value={{
              setUser
            }}
        >
            <LoginScreen />
        </UserContext.Provider>
    );
    
    test('Debe de renderizar el componemte correctamente', () => {
         
       expect(wrapper).toMatchSnapshot(); 
    });

    test('Debe de llamar correctamente a la funcion setUser con los agumentos esperados', () => {
        wrapper.find('button').simulate('click');
        
        expect(setUser).toHaveBeenCalledTimes(1);
        expect(setUser).toHaveBeenCalledWith(expect.any(Object));
        expect(setUser).toHaveBeenCalledWith({
            name:testUser.name,
            id:testUser.id,
        });
    });
});
