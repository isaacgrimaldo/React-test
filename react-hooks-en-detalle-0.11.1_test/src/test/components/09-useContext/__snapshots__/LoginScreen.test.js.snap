// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del LoginScreen.js Debe de renderizar el componemte correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
